sdks/java/core/src/main/java/org/apache/beam/sdk/io/UnboundedSource.java

```diff
@@ -93,6 +93,21 @@ public boolean requiresDeduping() {
     return false;
   }
 
+  /**
+   * If isOffsetDeduplication returns true, then the UnboundedSource needs to
+   * provide the following:
+   * 
+   * <ul>
+   *   <li>UnboundedReader which provides offsets that are unique for each
+   *       element and lexicographically ordered.</li>
+   *   <li>CheckpointMark which provides an offset greater than all elements
+   *       read and less than or equal to the next offset that will be read.</li>
+   * </ul>
+   */
+  public boolean isOffsetDeduplication() {
```

Comment:
maybe providesRecordOffsets? supportsOffsetBasedDeduping?

trying to think of something that is easy to read when in code

Reply:
Updated to offsetBasedDeduplicationEnabled